crates/primitives/src/chain/spec.rs

```diff
@@ -913,6 +916,14 @@ impl ChainSpec {
             C::Goerli => Some(goerli_nodes()),
             C::Sepolia => Some(sepolia_nodes()),
             C::Holesky => Some(holesky_nodes()),
+            #[cfg(feature = "optimism")]
+            C::Base | C::Optimism => Some(op_nodes()),
+            #[cfg(feature = "optimism")]
+            C::BaseGoerli |
+            C::BaseSepolia |
+            C::OptimismGoerli |
+            C::OptimismSepolia |
+            C::OptimismKovan => Some(op_testnet_nodes()),
```

Comment:
is this correct?

this uses the same nodes for multiple different chains?

Reply:
added the differentiation